python/cudf/cudf/core/frame.py

```diff
@@ -127,6 +127,10 @@ def _from_data(cls, data: MutableMapping):
         Frame.__init__(obj, data)
         return obj
 
+    @_cudf_nvtx_annotate
```

Comment:
Should this be a `@classmethod` like `_from_columns`?
```suggestion
    @classmethod
    @_cudf_nvtx_annotate
```

Reply:
No, this should be an instance method like `_from_columns_like_self`. Note the "like self" part, which indicates that we want to preserve some properties of `self`. However, I see one potential cause of your confusion, which is that I have the parameter named `cls` instead of `self`. Will fix that.

Reply:
Hang on -- this function just calls a `classmethod`, `cls._from_data`. If an instance method only calls a classmethod, it should also be a classmethod. OR this should say `self.__class__._from_data(data)`.

Reply:
Apologies if I'm not seeing something here. What properties of `self` are being retained?

Reply:
Great. I don’t _really_ care if we call through the class interface, since the lookup does cost something and is not necessary. I think this answers all my questions, I’ll double check and approve. Thanks!